config/collector_definitions.yaml

```diff
@@ -21,12 +21,12 @@
 ---
 openshift:
   default:
```

Comment:
Lets remove the endpoint role and `sources_per_deployment` concept for now.

We're not really using them and don't have a clear usecase for either until we get new source types.
At that point we'll probably have to refactor the collectors to read auth from a configmap anyway so adding this back in will be the least of our concerns.